tsig.go

```diff
@@ -39,6 +39,14 @@ type TSIG struct {
 	OtherData  string `dns:"size-hex:OtherLen"`
 }
 
+// TsigProvider provides the API to plug-in a custom TSIG implementation.
+type TsigProvider interface {
```

Comment:
This should be moved above `type TSIG struct { ... }` so it doesn't sit before the type's methods.

Reply:
Good point, sorted.